lib/Backend/GlobOpt.cpp

```diff
@@ -4844,6 +4845,15 @@ GlobOpt::CollectMemOpInfo(IR::Instr *instr, Value *src1Val, Value *src2Val)
             }
         }
     }
+    FOREACH_INSTR_IN_RANGE(chkInstr, instrBegin->m_next, instr->m_prev)
```

Comment:
To be safe, I think we should have all the conditions in the default case checked for instructions in instrBegin->m_next to instr->m_prev range.